nodejs/eks/nodegroup.ts

```diff
@@ -1837,56 +1851,38 @@ function createMNGCustomLaunchTemplate(
         : undefined;
 
     const serviceCidr = getClusterServiceCidr(core.cluster.kubernetesNetworkConfig);
+    const clusterMetadata = {
+        name: args.clusterName || core.cluster.name,
+        apiServerEndpoint: core.cluster.endpoint,
+        certificateAuthority: core.cluster.certificateAuthority.data,
+        serviceCidr,
+    };
+
     const customUserDataArgs = {
         kubeletExtraArgs: args.kubeletExtraArgs,
         bootstrapExtraArgs: args.bootstrapExtraArgs,
+        bottlerocketSettings: args.bottlerocketSettings,
     };
     let userData: pulumi.Output<string> | undefined;
     if (requiresCustomUserData(customUserDataArgs)) {
         userData = pulumi
-            .all([
-                core.cluster.name,
-                core.cluster.endpoint,
-                core.cluster.certificateAuthority.data,
-                args.clusterName,
-                os,
-                args.labels,
-                taints,
-                serviceCidr,
-            ])
-            .apply(
-                ([
-                    clusterName,
-                    clusterEndpoint,
-                    clusterCertAuthority,
-                    argsClusterName,
-                    os,
+            .all([clusterMetadata, os, args.labels, taints, args.bottlerocketSettings])
+            .apply(([clusterMetadata, os, labels, taints, bottlerocketSettings]) => {
+                const userDataArgs: ManagedNodeUserDataArgs = {
+                    nodeGroupType: "managed",
+                    kubeletExtraArgs: args.kubeletExtraArgs,
+                    bootstrapExtraArgs: args.bootstrapExtraArgs,
                     labels,
                     taints,
-                    serviceCidr,
-                ]) => {
-                    const clusterMetadata = {
-                        name: argsClusterName || clusterName,
-                        apiServerEndpoint: clusterEndpoint,
-                        certificateAuthority: clusterCertAuthority,
-                        serviceCidr,
-                    };
-
-                    const userDataArgs: ManagedNodeUserDataArgs = {
-                        nodeGroupType: "managed",
-                        kubeletExtraArgs: args.kubeletExtraArgs,
-                        bootstrapExtraArgs: args.bootstrapExtraArgs,
-                        labels,
-                        taints,
-
-                        // TODO: expose this as an option for managed node groups
-                        userDataOverride: undefined,
-                    };
-
-                    const userData = createUserData(os, clusterMetadata, userDataArgs, parent);
-                    return Buffer.from(userData, "utf-8").toString("base64");
-                },
-            );
+                    bottlerocketSettings,
+
+                    // TODO: expose this as an option for managed node groups
```

Comment:
Looks like a pre-existing TODO lacking a matching ticket. We usually have `TODO[pulumi/pulumi-eks#12345]: description` format for these.